templates/sync-function/time-module.js

```diff
@@ -0,0 +1,212 @@
+function timeModule() {
+  return {
+    isIso8601DateTimeString: isIso8601DateTimeString,
+    isIso8601DateString: isIso8601DateString,
+    isIso8601TimeString: isIso8601TimeString,
+    isIso8601TimeZoneString: isIso8601TimeZoneString,
+    compareTimes: compareTimes,
+    compareDates: compareDates,
+    compareTimeZones: compareTimeZones
+  };
+
+  // Check that a given value is a valid ISO 8601 format date string with optional time and time zone components
```

Comment:
From this point on, the functions are largely unchanged from how they appeared in the validation module.